science/providers/python_build_standalone.py

```diff
@@ -347,7 +347,7 @@ def create(cls, identifier: Identifier, lazy: bool, config: Config) -> PythonBui
         # For a given release (optional config parameter), get metadata.
         release_data = fetch_json(release_url, ttl=ttl)
 
-        release = release_data["name"]
+        release = release_data["tag_name"]
```

Comment:
Actually, this fixes non-latest as well. In that case I could have been using `config.release` but this release grab happens regardless since `"name"` `"tag_name"` is always set:
+ https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release
+ https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-a-release-by-tag-name